packages/web-components/src/components/va-search-input/va-search-input.tsx

```diff
@@ -0,0 +1,402 @@
+import {
+  Component,
+  Element,
+  Fragment,
+  Host,
+  h,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+
+@Component({
+  tag: 'va-search-input',
+  styleUrl: 'va-search-input.css',
+  shadow: true,
+})
+export class VaSearchInput {
+  @Element() el: HTMLElement;
+  inputRef!: HTMLInputElement;
+
+  /**
+   * An array of suggestions with suggested text bolded
+   */
+  @State() formattedSuggestions: string[] = [];
+
+  /**
+   * A boolean indicating whether listbox is open or closed
+   */
+  @State() isListboxOpen: boolean;
+
+  /**
+   * Text displayed inside the search button
+   */
+  @Prop() buttonText?: string;
+
+  /**
+   * The aria-label for search input and button. Default is 'Search'.
+   */
+  @Prop() label?: string = 'Search';
+
+  /**
+   * An array of strings containing suggestions to be displayed in listbox.
+   * This component displays up to 5 suggestions.
+   */
+  @Prop() suggestions?: any;
+
+  /**
+   * The value of the input field
+   */
+  @Prop() value?: string = '';
+
+  /**
+   * If suggestions are provided, then format suggestions and open the listbox.
+   * Limits suggestions to 5 and sorts them.
+   */
+  componentDidLoad() {
+    if (!Array.isArray(this.suggestions) || !this.suggestions?.length) return;
+    this.updateSuggestions(this.suggestions);
+  }
+
+  /**
+   * If suggestions are provided, then format suggestions and open the listbox.
+   * Limits suggestions to 5 and sorts them.
+   */
+  @Watch('suggestions')
+  watchSuggestionsHandler(newSuggestions: string[]) {
+    if (!Array.isArray(newSuggestions)) return;
```

Comment:
I think the typechecking on `newSuggestions: string[]` would prevent this from happening.

Reply:
This check is required because TypeScript doesn't check types on runtime.